lib/msal-common/src/url/UrlString.ts

```diff
@@ -61,7 +61,7 @@ export class UrlString {
         }
 
         // Throw error if URI or path segments are not parseable.
-        if (!components.HostNameAndPort || !components.PathSegments || components.PathSegments.length < 1) {
+        if (!components.HostNameAndPort || !components.PathSegments) {
```

Comment:
Maybe also want to get rid of the `PathSegments`? Not sure if this will be thrown or not in a case where the url is just "https://example.com" doesn't have a trailing slash. Would be good to test this.

Reply:
I tested this, even without the trailing `/` it doesn't throw. There is also a function (`canonicalizeUrl`) called in the constructor to normalize a `UrlString` with the trailing `/` and lowercase letters.

Reply:
Also added a unit test for this.